app/controllers/shifts_controller.rb

```diff
@@ -1,4 +1,5 @@
 class ShiftsController < ApplicationController
+  before_action :authenticate_user!, only: %i[ index show edit update destroy ]
```

Comment:
@saradotlog I added this `before_action` to the `Shifts` controller, so that the app checks for if the user is logged in before executing any of the methods in the array `%i[ index show edit update destroy ]`